### DISCUSSION TITLE
How do I add structure (year/month) to blog URLs?

### DISCUSSION THREAD

**Discussion by cellio** [datenstrom/community#696] [Ask a question] [answered]
I'm using the blog extension and would like URLs to be of the form: `blog/2021/09/some-title`.  I found a previous thread about whether people think that's a good idea, but I do sometimes have duplicate titles, for example when I write about the same topic (like a holiday) in different years, and I want to avoid collisions.  Further, I want people who copy and share a link to have that age indicator right there in the URL.

Following the documentation for the blog extension, I set the following in `yellow-system.ini`:

    BlogLocation: /blog/
    BlogNewLocation: /blog/@year/@month/@title

I've tried it without the month, to exactly match the example in the documentation.

My source files are in `2-blog` with names like `2021-09-12-some-title.md`.  Wondering if the content structure had to match the desired URL structure, I tried creating a `2021` directory and moving one of my source files there to test.  That produced a 404, so it looks like Yellow wants all the files to be in the top-level directory.

I don't care whether the source files are all in the same directory or in date-based subdirectories; I'll do whichever the tool requires.

In the source files, I set `Published` using ISO format, which seems to work as far as ordering and the archive are concerned.  For example:

    Title: something
    Published: 2021-09-12

I don't think I need to add `Year` and `Month` values explicitly (I assume those are parsed out of the date), but I tried it to see and it didn't make a difference.

I feel like I'm missing an important clue, but I've read the docs and tried these experiments and I'm at a loss for what to try next.  Any advice?

**Comment by markseuffert** [comment, marked as answer]
Your blog settings are correct, you probably need to adjust the content structure.

Have a look at [#239](https://github.com/datenstrom/yellow/discussions/239#discussioncomment-244744), example 3 should explain the setup your want.

**Reply by cellio** [reply]
Thanks.  For testing purposes, I simplified this to:

yellow-system.ini:
    BlogLocation: /blog/
    BlogNewLocation: /blog/@year/@title

With the following directory structure:

    content
        2-blog
            page.md
            2020
                2020-09-30-blah-blah-blah.md
                ...
            2021
                2021-05-04-something.md
                2021-09-12-something-else.md
                ...

When I visit the blog page all is well, but when I visit an individual blog post I get a URL of the form `example.com/blog/something`, and a "file not found" error.

I think I've replicated the structure in the post you linked, which is silent about page.md.  Do I need to put that somewhere other than the top-level directory?  Is there something else I've gotten wrong?

Thanks for the help!

**Reply by cellio** [reply]
And now, a few hours later, it's working.  Caching, maybe?  I didn't think to force a browser cache flush.

**Reply by markseuffert** [reply]
Usually a reload should be enough when playing with content structure. Sounds like [Murphy's law](https://en.wikipedia.org/wiki/Murphy%27s_law). 😁

**Reply by cellio** [reply]
Indeed!  My previous updates happened immediately, but they weren't *structural* changes, and I didn't think to insist.  Sorry for the extra noise, and thanks for the help!  I have the by-year structure working now (and might leave it that way instead of adding the month layer).

**Reply by markseuffert** [reply]
Thanks for your feedback, much appreciated. 👍

**Reply by pftnhr** [reply]
Hi @markseuffert! On my Site the blog is the start page and I can't get the folder for the year to be created in `1-home`. Instead, it is always created next to it, like this:
```
content
   1-home
      page.md
   3-tags
   2021
      09
         2021-09-21-fika-is-good-for-you.md
      10
         2021-10-01-fika-is-good-for-you.md
         2021-10-01-made-for-people.md
         2021-10-11-blog-page.md
```
Is there a way to make this work or is this not foreseen?

**Reply by markseuffert** [reply]
It looks correct to me. Please note that `content/1-home` folder must not contain subfolders. See also datenstrom/community#159.
